binary_transparency/firmware/internal/ftmap/pipeline.go

```diff
@@ -115,29 +148,50 @@ func (b *MapBuilder) getLogEnd(requiredEntries int64) (int64, []byte, error) {
 	return requiredEntries, golden, nil
 }
 
-// firmwareLogEntry is a parsed version of InputLogLeaf.
-type firmwareLogEntry struct {
-	Index    int64
-	Firmware api.FirmwareMetadata
+// partitionFn partitions the statements by type.
+// If you change the implementation here, then the cardinality needs to be
+// changed where this function is applied.
```

Comment:
Just a thought, you could have an explicit set of consts for the partitions like below and avoid/technically enforce the "if you ___, then don't forget to ___" :

```golang
const (
 FirmwareMetaPartition = iota
 MalwareStatementPartition
 // Add new partitions here
 ...
 MaxPartitions
)
```

and use `MaxPartitions` in the `beam.Partition` call directly.

Reply:
This is super neat, and I forgot Go could do enums for some reason. Done!